### PR TITLE
Generic client implementation

### PR DESCRIPTION
**Pull request by prempador**
- Removed old KassensichV example
- Added a generic client implementation
- Added test cases for new implementation
- Updated .gitignore
- Added submodule "sma" 
- Reworked readme.md
- Added code examples

### CODE REVIEW

File: kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift [hunk 1]
```diff
@@ -0,0 +1,332 @@
+//
+//  kassensichv_client_ios_tests.swift
+//  kassensichv_client_ios_tests
+//
+//  Created by Benjamin Müllner on 07.11.19.
+//  Copyright © 2019 fiskaly. All rights reserved.
+//
+
+import XCTest
+import kassensichv_client_ios
```
Comment (pgaubatz) [inline kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift]: can we rename that to `FiskalyKassensichvClient` to be consistent with current naming schema (`FiskalyKassensichvSma`)?

File: kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+class kassensichv_client_ios_tests: XCTestCase {
+    
+    let expectationTSS = XCTestExpectation(description: "Finish TSS Clall to API")
+    let expectationClient = XCTestExpectation(description: "Finish Client Call to API")
+    let expectationTX = XCTestExpectation(description: "Finish TX Call to API")
+    let expectationTX2 = XCTestExpectation(description: "Finish TX2 Call to API")
+    let expectationExport = XCTestExpectation(description: "Finish Export Call to API")
+    let expectation = XCTestExpectation(description: "Finish Client Call to API")
+    let client = {
```
Comment (pgaubatz) [inline kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift]: can't we simplify to:
```
let client = Client(
    apiKey: ProcessInfo.processInfo.environment["api_key"]!,
    apiSecret: ProcessInfo.processInfo.environment["api_secret"]!)
}
```

File: readme.md [hunk 3]
```diff
@@ -1,25 +1,88 @@
...
-- and how the SMA is used to intercept and sign transaction-related requests.
+First of all, you have to initialize the required git submodule(s) using:
 
-This **should be** the starting point for developing a reusable Swift or Objective-C based client that will eventually be hosted on https://github.com/fiskaly/fiskaly-kassensichv-client-ios, similar to clients for other languages and runtimes (e.g. https://github.com/fiskaly/fiskaly-kassensichv-client-node).
+```
+$ git submodule update --init
+```
 
-## How to get this example running?
+After that you need to extract the .tgz file to access the sma-Framework.
```
Comment (pgaubatz) [inline readme.md]: sma -> SMA

File: readme.md [hunk 4]
```diff
@@ -1,25 +1,88 @@
...
-![](screenshot-1.png)
+1. Select the project file from the project navigator on the left side of the project window.
+ 
+2. Select the target for where you want to add frameworks in the project settings editor.
+ 
+3. Select the “Build Phases” tab, and click the small triangle next to “Link Binary With Libraries” to view all of the frameworks in your application.
+ 
+4. To Add frameworks, click the “+” below the list of frameworks.
+
+5. Add Other - Add Files - Select extracted sma-Framework
```
Comment (pgaubatz) [inline readme.md]: sma -> SMA

File: readme.md [hunk 5]
```diff
@@ -1,25 +1,88 @@
...
+
+If you don't have an account on the [Fiskaly Dashboard](https://dashboard.fiskaly.com/) already you will need to create one and create an API-Key and -Secret pair.
```
Comment (pgaubatz) [inline readme.md]: Fiskaly -> fiskaly

File: readme.md [hunk 6]
```diff
@@ -1,25 +1,88 @@
...
+## Working with the client
+
+Currently the client takes your parameters and handles the HTTPRequest with the KassensichV API. 
+
+### Creating a client 
+
+```Swift
+import kassensichv_client_ios
+
+let client = {
```
Comment (pgaubatz) [inline readme.md]: simplify here as well (if possible)

File: readme.md [hunk 7]
```diff
@@ -1,25 +1,88 @@
...
+If you want to run the tests provided, you need to first add your API-Credentials as Environment-Variables or directly in the code. As soon as you have done that you can build and run the tests (⌘B ⌘U).
+
+## Working with the client
+
+Currently the client takes your parameters and handles the HTTPRequest with the KassensichV API. 
+
+### Creating a client 
+
+```Swift
+import kassensichv_client_ios
```
Comment (pgaubatz) [inline readme.md]: fix namespace as well (if possible)

**Review by pgaubatz — CHANGES_REQUESTED** [review]

File: kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift (hunk 1 above)
Comment (prempador) [reply kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift]: Renamed it 3b0f6ec

File: kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift (hunk 2 above)
Comment (prempador) [reply kassensichv-client-ios-tests/kassensichv_client_ios_tests.swift]: Simplified that 3b0f6ec

File: readme.md (hunk 3 above)
Comment (prempador) [reply readme.md]: Renamed it 3b0f6ec

File: readme.md (hunk 4 above)
Comment (prempador) [reply readme.md]: Renamed it 3b0f6ec

File: readme.md (hunk 5 above)
Comment (prempador) [reply readme.md]: Renamed it 3b0f6ec

File: readme.md (hunk 6 above)
Comment (prempador) [reply readme.md]: Simplified that 3b0f6ec

File: readme.md (hunk 7 above)
Comment (prempador) [reply readme.md]: Renamed it 3b0f6ec

**Review by pgaubatz — APPROVED** [review]

**Pull request merged**